assignment/tool/src/webapp/js/studentViewSubmission.js

```diff
@@ -31,3 +35,152 @@ ASN_SVS.undoCancel = function()
+        error: onError || $.noop,
+        complete: onComplete || $.noop
+    });
+};
+
+//SAK-43155
+ASN.switchTimesheetTab = function (source) {
+    if (['tabAssignment', 'tabTimeSheet'].includes(source)) {
+        document.getElementById('tabAssignmentCurrent').classList.toggle('hidden');
+        document.getElementById('tabAssignmentRef').classList.toggle('hidden');
+        document.getElementById('tabTimeSheetRef').classList.toggle('hidden');
+        document.getElementById('tabTimeSheetCurrent').classList.toggle('hidden');
+        document.getElementById('StudentAssignmentCurrent').classList.toggle('hidden');
+        document.getElementById('StudentTimesheetCurrent').classList.toggle('hidden');
+    }
+};
+
+//SAK-43155
+ASN.tsHandleAjaxAddSuccess = function (data) {
+    if (data.error && data.error.message) {
+        const button = document.getElementById("btnTimesheetAdd");
+        button.classList.remove("spinButton");
+        button.disabled = false;
+        const alertTsheetAddRecord = document.getElementById("alertTsheetAddRecord");
+        alertTsheetAddRecord.classList.toggle('hidden');
+        alertTsheetAddRecord.innerHTML= window.i18nWlogTab[data.error.message];
+    } else {
+        ASN.submitForm( 'addSubmissionForm', 'view', null, null );
+    }
+};
+
+//SAK-43155
+ASN.tsHandleAjaxRemoveSuccess = function (data) {
+    if (data.error && data.error.message) {
+        const button = document.getElementById("btnTimesheetDelete");
+        button.classList.remove("spinButton");
+        button.disabled = false;
+        const alertTsheetDelRecord = document.getElementById("alertTsheetDelRecord");
+        alertTsheetDelRecord.classList.toggle('hidden');
+        alertTsheetDelRecord.innerHTML= window.i18nWlogTab[data.error.message];
+    } else {
+        ASN.submitForm( 'addSubmissionForm', 'view', null, null );
+    }
+};
+
+//SAK-43155
+ASN.tsAddHandleAjaxError = function (xhr) {
+    const button = document.getElementById("btnTimesheetAdd");
+    button.classList.remove("spinButton");
+    button.disabled = false;
+    alert('Error: ' + xhr.status);
+    console.error("Ajax call error when add time sheet register.");
+};
+
+//SAK-43155
+ASN.tsRemoveHandleAjaxError = function (xhr) {
+    const button = document.getElementById("btnTimesheetDelete");
+    button.classList.remove("spinButton");
+    button.disabled = false;
+    alert('Error: ' + xhr.status);
+    console.error("Ajax call error when remove time sheet register.");
+};
+
+//SAK-43155
+ASN.checkTimesheetRecord = function () {
+    const selected = document.querySelectorAll("input[name='selectedTimesheet']:checked").length > 0;
+    document.getElementById("btnTimesheetDelete").disabled = !selected;
+    document.getElementById("btnTimesheetDelete").className = (selected ? "btn btn-primary active" : "");
+};
+
+//SAK-43155
+ASN.checkTimesheetSpent = function (totalTimeRecord) {
+    if (document.getElementById("timeTimesheet").checked) {
```

Comment:
```suggestion
    if (document.getElementById("timeTimesheet")?.checked) {
```

Reply:
optional chaining = la leche

Reply:
@adrianfish it is indeed!!! But we're not using yet because we have some teacheres with Linux Mint and Chrome 70 :'(